_blog-src/_posts/2016-04-04-locked-down-network.adoc

```diff
@@ -0,0 +1,79 @@
+
+We've carefully designed apiman to be fully featured and easily configured when no Internet access is available; providing a great deal of flexibility and eschewing any "off-site only" functionality.
+
+So, if you're looking for API management in a locked-down network or Internet-free environment, read on!
+
+////
+Use this pattern to avoid asciidoctor rendering the HTML comment.
+<!--more-->
+////
+
+toc::[]
+
+== Apiman Plugins
+
+Apiman's extensibility is one of its key strengths, and it's extremely easy to write http://www.apiman.io/latest/developer-guide.html#_plugins[plugins] that can do almost anything. There are two default behaviours we are likely to want to configure when operating within a restricted network.
+
+=== Distribution and Management
+
+Apiman plugins are, by default loaded via http://central.sonatype.org/[The Central Repository] (often known as Maven Central), which is a fantastically convenient in most situations. All of apiman's components (manager, gateways, etc), will pull down the necessary plugins as and when they need them.
+
+However, without Internet access this clearly won't work as intended. We have three good options which you can configure in `apiman.properties`:
+
+```properties
+apiman.plugins.repositories=https://repo1.example.local/our-internal-repo/,http://repository.example.org/nexus/content/groups/public/ <1>
+
+apiman.plugins.repositories=file:///mnt/nfs/my/shared/maven/repo <2>
+
+apiman.plugins.repositories=<3>
+```
+<1> Use an internal https://maven.apache.org/repository-management.html#Available_Repository_Managers[repository or mirror] to deploy your own (and community) plugins to. This is likely the best option for many enterprises who may already have such infrastructure. Join multiple repositories with commas.
+<2> Mount a repository to a shared volume or filesystem on every machine running apiman.
+<3> Copy a valid Maven formatted repository containing your plugins to `~/.m2`, where apiman will look by default.
+
+TIP: *Advanced:* You can replace the https://github.com/apiman/apiman/blob/master/gateway/engine/core/src/main/java/io/apiman/gateway/engine/impl/DefaultPluginRegistry.java[default registry] entirely with an implementation of your own and set it via `apiman-gateway.plugin-registry` in `apiman.properties`.
+
+=== Registry (Plugin Catalogue)
+
+Our community releases contain a plugin registry which downloads a curated list of plugins. This lets the API Manager you have a nice prepopulated list of plugins to install, rather than having to type the coordinates manually. To preserve this functionality you can simply download the registry file and make it available internally:
+
+```properties
+apiman-manager.plugins.registries=http://rawgit.com/apiman/apiman-plugin-registry/master/registry.json <1>
+
+apiman-manager.plugins.registries=file:///path/to/my/registry.json <2>
```

Comment:
TIP: when running in wildfly/eap you can use appropriate platform system properties to make the config more portable:

```
apiman-manager.plugins.registries=file:///${jboss.server.config.dir}/apiman-plugin-registry.json
```